### DISCUSSION TITLE
WebAuthn with HTMX anyone?

### DISCUSSION THREAD

**Discussion by hrstoyanov** [bigskysoftware/htmx#1568] [Q&A]
WebAuthn is the new way to do authentication in 2023 and there are some examples for doing this that are heavy on JSON APIs and Javascript. Does anyone have implementation the "htmx way"  that can share?

**Comment by yawaramin** [comment]
Htmx can't really do much for WebAuthn, it needs to use a bunch of JavaScript to run the client-side stuff. The closest you will get to an 'htmx example' is probably https://github.com/AnswerDotAI/fasthtml-example/blob/main/passkey/main.py

**Comment by hrstoyanov** [comment]
No worries. I implemented WebAuthn/Passkeys without HTMX. It is JS-heavy workflow, best to keep it that way. I requires tigth coupling between server and client to implement the 2-step use cases
